plugins/rh-rbac-backend/src/service/backend-plugin-ids-provider.ts

```diff
@@ -0,0 +1,61 @@
+import { findPaths } from '@backstage/cli-common';
+
+import * as lockfile from '@yarnpkg/lockfile';
+
+import fs from 'fs';
+
+// BackendPluginIDsProvider should retrieve list backend plugin from
+// the root yarn.lock.
+export class BackendPluginIDsProvider {
+  private readonly orgPrefixes = ['@backstage/', '@janus-idp/'];
+  private backendSuffixRegExp = new RegExp('-backend@.*$');
+
+  // Get list id for backend plugins.
+  public getPluginIds(): string[] {
+    /* eslint-disable-next-line no-restricted-syntax */
+    const paths = findPaths(__dirname);
+    const lockfilePath = paths.resolveTargetRoot('yarn.lock');
+    const file = fs.readFileSync(lockfilePath, 'utf8');
+    const lockFileRaw = lockfile.parse(file);
+
+    if (lockFileRaw.type !== 'success') {
+      throw new Error(
+        `Failed to parse yarn.lock file to get list backend plugins`,
+      );
+    }
+
+    return this.getBackendPluginIds(Object.keys(lockFileRaw.object));
+  }
+
+  private getBackendPluginIds(deps: string[]): string[] {
+    const backendPlugins = this.filterBackendPlugins(deps);
+
+    return this.pasePluginIds(backendPlugins);
+  }
+
+  private filterBackendPlugins(deps: string[]): string[] {
+    return deps.filter(dep => {
+      const isOrgPrefixed = this.orgPrefixes.some(orgPrefix =>
+        dep.startsWith(orgPrefix),
+      );
+      return isOrgPrefixed && this.backendSuffixRegExp.test(dep);
+    });
+  }
+
+  private pasePluginIds(deps: string[]) {
+    const prefixes = this.orgPrefixes.map(orgPrefix => `${orgPrefix}plugin-`);
```

Comment:
I am not sure about this method of scanning the lock file for backend plugins is 100% correct. This will at least miss the dynamic plugins. Also the backstage cli determines a backend plugin by looking at `backstage.role` on `package.json`.

Reply:
> I am not sure about this method of scanning the lock file for backend plugins is 100% correct.

I think most developers generates backstage plugins using cli. CLI follows ADR: https://backstage.io/docs/architecture-decisions/adrs-adr011#decision and developers should also follow this ADR

> Also the backstage cli determines a backend plugin by looking at `backstage.role` on package.json

While the idea of retrieving the package.json for all dependencies and searching for `backstage.role` is an option, I have reservations about its performance. Given the likelihood of numerous dependencies, I believe this approach may not meet our performance requirements effectively. But if you really want this solution, I can try.

> This will at least miss the dynamic plugins.

I agree. Dynamic plugins is a new staff for me and need some time to figure out we can list them too.